teamvault/apps/secrets/models.py

```diff
@@ -227,24 +227,47 @@ def get_data(self, user):
             category = AuditLogCategoryChoices.SECRET_READ
             log_message = _("{user} read '{name}'")
 
-        log(
-            log_message.format(
-                name=self.name,
-                user=user.username,
-            ),
-            actor=user,
-            category=category,
-            level='info',
-            secret=self,
-            secret_revision=self.current_revision,
-        )
+        if request:
+            timediff = datetime.now() - datetime.fromisoformat(
+                request.session["session_start"]).replace(
+                tzinfo=None)
+            timediff = timediff.seconds
+            sessions_tracked = request.session["sessions_tracked"]
+            if sessions_tracked == 0 or timediff >= sessions_tracked * 1800 or not get_otp_key:
+                log(
+                    log_message.format(
+                        name=self.name,
+                        user=user.username,
+                    ),
+                    actor=user,
+                    category=category,
+                    level='info',
+                    secret=self,
+                    secret_revision=self.current_revision,
+                )
+                request.session["sessions_tracked"] += 1
+        else:
+            log(
+                log_message.format(
+                    name=self.name,
+                    user=user.username,
+                ),
+                actor=user,
+                category=category,
+                level='info',
+                secret=self,
+                secret_revision=self.current_revision,
+            )
         self.current_revision.accessed_by.add(user)
         self.current_revision.save()
         self.last_read = now()
         self.save()
 
         f = Fernet(settings.TEAMVAULT_SECRET_KEY)
-        plaintext_data = f.decrypt(self.current_revision.encrypted_data.tobytes())
+        if get_otp_key and self.current_revision.encrypted_otp_key_data:
+            plaintext_data = f.decrypt(self.current_revision.encrypted_otp_key_data.tobytes())
+        else:
+            plaintext_data = f.decrypt(self.current_revision.encrypted_data.tobytes())
```

Comment:
Hier würde es sich anbieten `encrypted_data` in eine eigene Variable zu tun und das `f.decrypt(encrypted_data.tobytes())` dann nach dem if zu machen. Weniger noise und so.